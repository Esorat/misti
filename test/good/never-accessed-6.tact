fun test(): Int {
    let a: Int = 20; // read-only variable
    let b: Int = 22; // read-only variable
    let c: Int = 24;
    if (true) { // no `neverAccessed` false positive: all are used
        c = a + b;
    }
    return c;
}
