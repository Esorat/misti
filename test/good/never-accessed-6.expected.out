test/good/never-accessed-6.tact:2:5:
  1 | fun test(): Int {
> 2 |     let a: Int = 20; // read-only variable
          ^
  3 |     let b: Int = 22; // read-only variable
Read-only variable
Help: Consider creating a constant instead
See: https://nowarp.github.io/docs/misti/docs/detectors/ReadOnlyVariables
test/good/never-accessed-6.tact:3:5:
  2 |     let a: Int = 20; // read-only variable
> 3 |     let b: Int = 22; // read-only variable
          ^
  4 |     let c: Int = 24;
Read-only variable
Help: Consider creating a constant instead
See: https://nowarp.github.io/docs/misti/docs/detectors/ReadOnlyVariables